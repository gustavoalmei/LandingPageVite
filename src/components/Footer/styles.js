import { styled, css } from 'styled-components';
import { Container as TextComponet } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../../components/SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${TextComponet} {
      font-size: ${theme.fonts.sizes.xsmall};
    }

    & > ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
