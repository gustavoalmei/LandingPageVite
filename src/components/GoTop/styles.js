import { styled, css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background-color: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;

    right: 2rem;
    bottom: 2rem;
    z-index: 6;
    opacity: 0.7;
    border-radius: 50%;
  `}
`;
