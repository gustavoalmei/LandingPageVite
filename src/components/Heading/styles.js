import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${mediaFont(theme)}
  `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.large};
  }
`;

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1.withConfig({
  // withConfig permite configurar como o Styled Components manipula props específicas.
  shouldForwardProp: (prop) => !['colorDark', 'uppercase'].includes(prop),
  // shouldForwardProp é uma função que decide quais props serão passadas para o DOM. Nesse caso, estamos dizendo que colorDark não deve ser repassado.
})`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)}
    ${titleCase(uppercase)}
  `}
`;
