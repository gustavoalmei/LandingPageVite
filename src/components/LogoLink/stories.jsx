import LogoLink from '.';

export default {
  title: 'LogoLink',
  component: LogoLink,
  args: {
    text: 'LogoLink',
    imgLink: 'assets/images/logo.svg',
    link: 'http://localhost',
  },
  argTypes: {
    text: { type: 'string' },
  },
};

export const Text = (args) => {
  return (
    <div>
      <LogoLink {...args} />
    </div>
  );
};

export const Logo = (args) => {
  return (
    <div>
      <LogoLink {...args} />
    </div>
  );
};

Text.args = {
  imgLink: '',
};
