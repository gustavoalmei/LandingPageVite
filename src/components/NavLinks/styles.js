import { styled, css } from 'styled-components';

export const Container = styled.nav`
  ${({ theme }) => css`
    display: flex;
    flex-flow: row wrap;

    @media (max-width: 768px) {
      flex-flow: column wrap;
      align-content: center;
    }
  `}
`;
