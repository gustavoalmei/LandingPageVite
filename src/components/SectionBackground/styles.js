import { styled, css } from 'styled-components';

const setBackgroundColor = (theme) => css`
  background: ${theme.colors.primaryColor};
  color: ${theme.colors.white};
`;

export const Container = styled.div.withConfig({
  shouldForwardProp: (prop) => !['background'].includes(prop),
})`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    ${background && setBackgroundColor(theme)};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
