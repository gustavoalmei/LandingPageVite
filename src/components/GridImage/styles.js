import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    height: 100%;
    object-fit: cover;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.3) rotate(10deg);
    }
  `}
`;
