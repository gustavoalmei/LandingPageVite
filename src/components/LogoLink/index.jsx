import P from 'prop-types';
import { Heading } from '../Heading';
import * as S from './styles';

export default function LogoLink({ text, srcImg, link }) {
  return (
    <Heading size="small" uppercase>
      <S.Container href={link}>
        {!!srcImg && <img src={srcImg} alt={text} />}
        {!srcImg && text}
      </S.Container>
    </Heading>
  );
}

LogoLink.propTypes = {
  text: P.string.isRequired,
  srcImg: P.string,
  link: P.string.isRequired,
};
