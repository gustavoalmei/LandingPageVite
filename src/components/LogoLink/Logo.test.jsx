import { screen } from '@testing-library/react';
import { RenderTheme } from '../../styles/handleTheme';
import LogoLink from './index';

describe('<LogoLink/>', () => {
  test('Should be render text logo', () => {
    RenderTheme(<LogoLink link="#target" text="Olá mundo" />);
    const heading = screen.getByRole('heading', { name: 'Olá mundo' });

    expect(heading).toBeInTheDocument();
    expect(heading.firstChild).toHaveAttribute('href', '#target');
  });

  test('Should be render image logo', () => {
    RenderTheme(
      <LogoLink link="#target" srcImg="image.jpg" text="Olá mundo" />,
    );
    const heading = screen.queryByAltText('Olá mundo');
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveAttribute('src', 'image.jpg');
  });

  test('Should match ', () => {
    RenderTheme(
      <LogoLink link="#target" srcImg="image.jpg" text="Olá mundo" />,
    );
    expect(
      screen.getByRole('heading', { name: 'Olá mundo' }),
    ).toMatchSnapshot();
  });
});
