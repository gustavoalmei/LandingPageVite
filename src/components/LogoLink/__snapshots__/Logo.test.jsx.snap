// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink/> Should match  1`] = `
.c0 {
  color: #0A1128;
  font-size: 2.5rem;
  text-transform: uppercase;
}

.c1 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
}

.c1 >img {
  max-height: 3rem;
  width: 100%;
}

<h1
  class="c0"
>
  <a
    class="c1"
    href="#target"
  >
    <img
      alt="OlÃ¡ mundo"
      src="image.jpg"
    />
  </a>
</h1>
`;
